### DISCUSSION TITLE
LAB 05.01 Task 3: ML Metric vs Business Metric

### DISCUSSION THREAD

**Discussion by moralesluis26** [rramosp/ai4eng.v1#186] [05 - GESTIÓN DE MODELOS]
Saludos.
Tengo una duda similar a la que hay en un post anterior :
https://github.com/rramosp/ai4eng.v1/discussions/64#discussion-3599890
 
Obtengo las filas y columnas de la matriz pedida, pero no sé como calcular el NPL.
En el ejercicio anterior de ese mismo Lab, el NPL se obtenia de cada uno de los valores asociados a una predicción y de la señal original. Ahora con esta matríz, intenté sacar el promedio y sumar todos los datos del NPL asociados a cada una de las predicciones del array y no están correctas. 
De ante mano agradezco por la atención.

**Comment by Felipe-RA** [comment]
Saludos moralesluis26

Muestrame cómo lo estás intentando solucionar para poder explicarte mejor cómo podrías solucionarlo o qué puedes estar dejando de lado

**Reply by moralesluis26** [reply]
![imagen](https://user-images.githubusercontent.com/46792756/184792442-31614bfd-1b22-49b7-8838-f4f1efc05159.png)
Realemnete lo estoy haciendo muy manual. Lo que hago inicialmente es crear cada una de las filas del array que piden asegurando los porcentajes de acierto de 0.1 a 0.9. Cuando hago cada uno, calculo el PNL, y como tengo valores PNL para cada uno de los datos de predicción de cada fila del array, lo que hago es sumerlos y ese lo dejo como el valor representativo del PNL de la fila. Cuando evalúo, me arroja un resultado erróneo. Adjunto imágen de lo que he trabajo. 
De antemano le agradezco.

**Comment by Felipe-RA** [comment]
Hola moralesluis26

Entiendo el acercamiento que le estás dando, pero considero que presenta muchos inconvenientes para que llegues a la solución del problema.

Te recomiendo  el siguiente acercamiento:

Implementa el cálculo del pnl como una función, puedes llamarla `get_pnl()` o `calcular_pnl()`, de forma tal que reciba un `signal` y un `predictions`. Esto te permitirá calcular el pnl para cada fila.

Aunque no es necesario para este ejercicio, ¡incluso podrías ponerle el número de `trading_units` para hacerlo más generalizado, más cercano a una implementación real!

Puede verse como algo tal que así:

```
def get_pnl(signal, predictions, trading_units=1)
    ## tu codigo para calcular pnl
    return pnl
```

Ahora, para obtener el array 9x20 te recomiendo las siguientes funciones:

- `np.linspace()` para obtener un vector con los `accuracies`. Fijate en la [documentación](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html) para entender cómo esta función te entrega un vector con números espaciados sobre un intervalo definido.
- `np.zeros()` para obtener un vector con dimensiones dadas por una tupla (x,y) [en este caso 9x20]

Luego, puedes hacer un list comprehension para aplicar `get_pnl()` sobre las filas de pset

Me cuentas cómo te va!

**Reply by moralesluis26** [reply]
Listo! Muchas gracias, ya me funcionó.

**Reply by Felipe-RA** [reply]
Me alegra!

Si lo deseas, compártenos cómo lo solucionaste
